tests/handlers/github_test.py

```diff
@@ -0,0 +1,158 @@
+        org=org,
+        repo=repo,
+        ref=ref,
+    )
+
+    # https://docs.github.com/en/webhooks/using-webhooks/validating-webhook-deliveries#python-example
+    hash_object = hmac.new(
+        TEST_GITHUB_WEBHOOK_SECRET.encode("utf-8"),
+        msg=payload.encode("utf-8"),
+        digestmod=hashlib.sha256,
+    )
+    sig = "sha256=" + hash_object.hexdigest()
+
+    headers = {
+        "Accept": "*/*",
+        "Content-Type": "application/json",
+        "User-Agent": "GitHub-Hookshot/c9d6c0a",
+        "X-GitHub-Delivery": "d2d3c948-1d61-11ef-848a-c578f23615c9",
+        "X-GitHub-Event": "push",
+        "X-GitHub-Hook-ID": "479971864",
+        "X-GitHub-Hook-Installation-Target-ID": "804427678",
+        "X-GitHub-Hook-Installation-Target-Type": "repository",
+        "X-Hub-Signature-256": sig,
+    }
+
+    return GithubRequest(payload=payload, headers=headers)
+
+
+@pytest.mark.asyncio
+async def test_handle_webhook(
+    client: AsyncClient,
+    anon_client: AsyncClient,
+    respx_mock: respx.Router,
+    mocker: MockerFixture,
+) -> None:
+    configs = [
+        {
+            "name": "test-notebook",
+            "count": 1,
+            "user_spec": {"username_prefix": "testuser-notebook"},
+            "scopes": ["exec:notebook"],
+            "business": {
+                "type": "NotebookRunner",
+                "options": {
+                    "repo_url": "https://github.com/lsst-sqre/some-repo.git",
+                    "repo_branch": "main",
+                },
+            },
+        },
+        {
+            "name": "test-notebook-branch",
+            "count": 1,
+            "user_spec": {"username_prefix": "testuser-notebook-branch"},
+            "scopes": ["exec:notebook"],
+            "business": {
+                "type": "NotebookRunner",
+                "options": {
+                    "repo_url": "https://github.com/lsst-sqre/some-repo.git",
+                    "repo_branch": "some-branch",
+                },
+            },
+        },
+        {
+            "name": "test-other-notebook",
+            "count": 1,
+            "user_spec": {"username_prefix": "testuser-other-notebook"},
+            "scopes": ["exec:notebook"],
+            "business": {
+                "type": "NotebookRunner",
+                "options": {
+                    "repo_url": "https://github.com/lsst-sqre/some-other-repo.git",
+                    "repo_branch": "main",
+                },
+            },
+        },
+        {
+            "name": "test-non-notebook",
+            "count": 1,
+            "user_spec": {"username_prefix": "testuser-non-notebook"},
+            "scopes": ["exec:notebook"],
+            "business": {"type": "EmptyLoop"},
+        },
+    ]
+
+    # Don't actually do any business
+    mocker.patch.object(mobu.services.flock.Monkey, "start")  # type: ignore[attr-defined]
+    mocker.patch.object(mobu.services.flock.Monkey, "stop")  # type: ignore[attr-defined]
```

Comment:
Is there a better way to do this?

Reply:
Yeah, I think you can just use `unittest.mock.patch` in a context manager (probably as a yield fixture) to stub out those methods. That's usually what I do.